backend/src/main.cpp

```diff
@@ -47,10 +47,22 @@ int main(int argc, char *argv[])
     const size_t offset = 1;
     spdlog::info("Password offset: {}", offset);
 
+    // limit the number of leaked bytes to 4
+    if (offset > 4)
+    {
+        throw std::invalid_argument("Offset cannot be greater than 4.");
+    }
+
+    // build the database
     if (build)
     {
-        // create password table
-        db.execute("CREATE TABLE passwords (password TEXT);");
+        // create all tables from 0 to (2^8)^offset-1
+        for (int i = 0; i < std::pow(std::pow(2, 8), offset); i++)
+        {
+            std::string query_str = "table" + std::to_string(i);
+            db.execute("CREATE TABLE " + query_str + " (password TEXT);");
+            spdlog::info("Created table: {}", query_str);
```

Comment:
**suggestion**: This is too much logging for `info`. Could be `debug` or just removed.